Applications/FileIO/CsvInterface.cpp

```diff
@@ -22,6 +22,44 @@
 namespace FileIO {
 CsvInterface::CsvInterface() = default;
 
+std::vector<std::string> CsvInterface::getColumnNames(std::string const& fname,
+                                                      char delim)
+{
+    std::ifstream in(fname.c_str());
+
+    if (!in.is_open())
+    {
+        ERR("CsvInterface::readPoints(): Could not open file %s.",
+            fname.c_str());
+        return std::vector<std::string>();
+    }
+    std::string line;
+    if (!getline(in, line))
+        return std::vector<std::string>();
+
+    std::list<std::string> fields;
+    if (delim != '\n')
+        fields = BaseLib::splitString(line, delim);
+
+    if (fields.size() < 2)
+    {
+        std::array<char, 3> delims = {'\t', ';', ','};
+        for (char d : delims)
+        {
+            fields = BaseLib::splitString(line, d);
+            if (fields.size() > 1)
+            {
+                delim = d;
+                break;
+            }
+        }
+    }
+    std::vector<std::string> names;
+    names.reserve(names.size());
+    std::copy(fields.begin(), fields.end(), std::back_inserter(names));
+    return names;
```

Comment:
```suggestion
    return {begin(fields), end(fields)};
```
no need to do all the work by hand.